gldemo/plugin.cpp

```diff
@@ -363,20 +398,26 @@ class gldemo : public threadloop {
 		// Load/initialize the demo scene.
 
 		char* obj_dir = std::getenv("ILLIXR_DEMO_DATA");
-		if(obj_dir == NULL) {
+		if (obj_dir == nullptr) {
 			std::cerr << "Please define ILLIXR_DEMO_DATA." << std::endl;
-			abort();
+            std::exit(1);
```

Comment:
At faults, I think `abort` is better because it fires a `SIGABRT` signal, which can be caught by a debugger. `std::exit` and `std::terminate` cannot, AFAIK. I suppose this is why `assert` sends a `SIGABRT`.

The question is, should destructors be run if the exit is due to a fatal error? I think the answer is no. In the case of a fatal error, the output is likely "tainted" by the error. In the _special case_ that the operator does want it, they can replace that particular `abort` with an `std::exit`, but I contend this should not be the default,

Reply:
`abort` does not allow the C++ runtime to perform cleanup:
>  The difference between exit and abort is that exit allows the C++ run-time termination processing to take place (global object destructors get called), but abort terminates the program immediately. The abort function bypasses the normal destruction process for initialized global static objects. It also bypasses any special processing that was specified using the atexit function.
[Source](https://docs.microsoft.com/en-us/cpp/cpp/program-termination?view=msvc-160). It seems to me that `abort` could be useful for a debugging environment, though. Perhaps we can use a small utility that switches the `std::exit` to `abort` if `#ifdndef NDEBUG`?

Reply:
In terms of the destructors, we can be holding onto stateful resources (e.g., file handles) which should be closed normally if possible to avoid odd side-effects.

Reply:
That is, I think debugging via `SIGABRT` is the special case, given that the user will need to resolve the failure.